app/controllers/admin/merchants_controller.rb

```diff
@@ -0,0 +1,52 @@
+class Admin::MerchantsController < ApplicationController
+  before_action :fetch_current_merchant, only: [:edit, :show, :update]
+  before_action :fetch_merchant_collection, only: [:index]
+  before_action :fetch_new_merchant, only: [:new]
```

Comment:
curious what these lines of code are doing 😎

Reply:
For sure!  Basically what those are doing is calling the methods listed in the private methods at the bottom.  Then those methods get called with the actions in the array after the `only:`.  

In this case, it makes the most sense for the `fetch_current_merchant` method since the same instance variable is used in each of the three actions and it DRYs up the code ... instead of having to repeat the same line in each action that stores the Merchant object in the instance variable.  You could argue it is not really DRYing up the code for the `fetch_merchant_collection` and `fetch_new_merchant` methods (...well it isn't DRYing it up at all here! haha!), but I just went with it to be consistent.

You can check out a little more on it here:
- https://guides.rubyonrails.org/v5.2/action_controller_overview.html#filters